TesasSemble/contig_assembly.py

```diff
@@ -0,0 +1,45 @@
+def EdgePathtoNodePath(edgePath):
+    """Convert a path represented edges to a path represented by respective nodes"""
+    nodePath = []
+    for path in edgePath:
+        node_path = [path[0].node_a,path[0].node_b]
+        if len(path)>1:
+            for edge in path[1:]:
+                node_path.append(edge.node_b)
+        nodePath.append(node_path)
+    return nodePath
+
+def overlapBetweenStrings(string1,string2):
+    """Brute force find overlap between suffix of string1 and prefix of string 2"""
+    i = 0
+    suffString1 = string1
+    prefString2 = string2
+    while prefString2 != suffString1:
+        i += 1
+        suffString1 = string1[i:]
+        prefString2 = string2[0:len(suffString1)]
+    return len(prefString2)-1  
+
+def stringSpelledByNodePath(path,reads_map):
+    """String spelled by a path represented by node values"""
+    string = reads_map[int(str(path[0]))]
+    for i in range(1,len(path)):
+        string1 = reads_map[int(str(path[i-1]))] 
+        # this indexing may need to be changed depending on key values (are they ints?)
+        # if it is nodes instead of ints, can get ride of int(str(... and just put path[i-1]
+        string2 = reads_map[int(str(path[i]))]
+        index = overlapBetweenStrings(string1,string2)
+        string += string2[index+1:]
+    return string
+
+def contig_assembly(graph,reads_map):
+    """ 
+    input: DiGraph object
```

Comment:
reads_map should actually be a dictionary object from {node object : associated read}